### PR TITLE
blue/green: Update items in the transaction all at once

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR slightly refactors the `RenameSchema` op to update the durable transaction with the new items all at once instead of calling `tx.update_item(...)` for each item. `update_item(...)` iterates through the entire catalog every time it's called so it can be quite slow.

Creating two schemas with 100 views each, and then 500 views depending on the 100 views in a schema we exhibit the following performance.

`main`
```
materialize=> ALTER SCHEMA blue SWAP WITH green;
ALTER SCHEMA
Time: 37301.170 ms (00:37.301)
```

This PR
```
materialize=> ALTER SCHEMA blue SWAP WITH green;
ALTER SCHEMA
Time: 470.736 ms
```

### Motivation

_Greatly_ improves the speed of `ALTER SCHEMA ... SWAP WITH ...`

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Performance improvement for `ALTER SCHEMA ... SWAP WITH ...`

### CODE REVIEW

File: src/adapter/src/catalog.rs [hunk 1]
```diff
@@ -2504,14 +2503,11 @@ impl Catalog {
 
                         // Update the catalog storage and Builtin Tables.
```
Comment (jkosh44) [inline src/adapter/src/catalog.rs:2504]: Might want to update this comment since it's not 100% accurate. We're not updating the catalog storage yet.

**Review by jkosh44 — APPROVED** [review]
Nice!

File: src/adapter/src/catalog.rs (hunk 1 above)
Comment (ParkMyCar) [reply src/adapter/src/catalog.rs]: Good call, updated!

**Review by benesch — COMMENTED** [review]
Nice! Can we add a test for this? Maybe in test/limits? No need to block this PR on it. And perhaps @MaterializeInc/testing have some cycles to assist.

**Comment by ParkMyCar** [comment]
> Nice! Can we add a test for this? Maybe in test/limits? No need to block this PR on it. And perhaps @MaterializeInc/testing have some cycles to assist.

Will do! I'll follow up with the testing team about this

**Comment by nrainer-materialize** [comment]
> Will do! I'll follow up with the testing team about this

The test will be added with https://github.com/MaterializeInc/materialize/pull/26377.

**Pull request merged**